Automattic/pocket-casts-ios
# Use Episode IDs from the sync task response instead of our local ones

Uses the UDIDs of the episodes from the sync response instead of re-querying for the unsynced episodes. This fixes an issue where users with 2000+ episodes in a background sync could send the same episodes during background sync repeatedly.

* Adds a `useSyncResponseEpisodeIDs` feature flag in case we need to disable
* Adds duplicate method for marking episodes synced with their `uuid` instead of the local `id`
* Updates `processServerData` to use these `uuids` when handling the server response

### Unit Tests

* Adds unit tests to test the new update behavior from the server response 

## To test

* Build and run the app
* Subscribe to some podcasts
* Add several episodes
* Mark several episodes as archived
* Refresh via the profile or kill and relaunch the app
* Make sure the sync occurs. You should see some logs in the console:

```
SyncTask: sending 6 changed items to the server
```

## Checklist
